### PR TITLE
[macsec/test_dataplane]: Add macsec counter clear test

### LINKED ISSUE #13347: [Test Gap] Test to check sonic-clear macsec
### Test Gap Description

sonic-mgmt test to check issues like - https://github.com/sonic-net/sonic-buildimage/issues/19311
Test sonic macsec counters and the command `sonic-clear counters` works on macsec counters as well

### Test Plan

_No response_

### Test environment

_No response_

### Attach files (if any)

_No response_

### PR DESCRIPTION
**Pull request by liamkearney-msft**
### Description of PR
Add test for macsec counters clear command.

Summary:
Fixes #13347

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ x] 202405

#### What is the motivation for this PR?
Fix test gap by adding sonic-clear macsec tests
#### How did you verify/test it?
Ran locally on t2 chassis, test passes

### CODE REVIEW

File: tests/macsec/test_dataplane.py [hunk 1]
```diff
@@ -198,3 +186,52 @@ def test_counters(self, duthost, ctrl_links, upstream_links, rekey_period, wait_
...
+
+        egress_cleared_counters, ingress_cleared_counters = self._get_counters(duthost, up_ports)
+
+        for counter in egress_end_counters:
+            if counter.startswith('SAI_MACSEC_SA_STAT'):
+                if egress_end_counters[counter] != 0:
+                    assert egress_cleared_counters[counter] < egress_end_counters[counter]
+            else:
+                # Non stats should not reset
+                # TODO: Currently XPN is reset - uncomment this when fixed
```
Comment (liamkearney-msft) [inline tests/macsec/test_dataplane.py:225]: this needs this PR to be enabled : https://github.com/sonic-net/sonic-buildimage/pull/20654

**Review by arlakshm — APPROVED** [review]

**Comment by rlhui** [comment]
@liamkearney-msft  please resolve conflicts? Thanks.

**Comment by liamkearney-msft** [comment]
@rlhui conflicts have been resolved

**Review by arlakshm — APPROVED** [review]

**Comment by arlakshm** [comment]
@judyjoseph, please help signoff this change

**Review by judyjoseph — APPROVED** [review]

**Comment by bingwang-ms** [comment]
@liamkearney-msft Is the new test `test_clear_counters` for T2 only or for T0 and T2? Can you paste the test result to the PR since it's a new test?

**Comment by liamkearney-msft** [comment]
Hi @bingwang-ms , this test should be topo agnostic. 
Ive run these tests manually on both arista 7800 & nokia 7250, and it passes : 
```
04:10:41 __init__.pytest_terminal_summary         L0067 INFO   | Can not get Allure report URL. Please check logs
=============================================================== 1 passed, 1 warning in 953.59s (0:15:53) ===============================================================
DEBUG:tests.conftest:[log_custom_msg] item: <Function test_clear_counters[MACSEC_PROFILE]>
```

**Pull request merged**